src/client/pkg/grpcutil/addr.go

```diff
@@ -0,0 +1,127 @@
+package grpcutil
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+const (
+	// DefaultPachdNodePort is the pachd kubernetes service's default
+	// NodePort.Port setting
+	DefaultPachdNodePort = 30650
+
+	// DefaultPachdPort is the pachd kubernetes service's default
+	// Port (often used with Pachyderm ELBs)
+	DefaultPachdPort = 650
+)
+
+var (
+	// ErrNoPachdAddress is returned by ParsePachdAddress when the input is an
+	// empty string
+	ErrNoPachdAddress = errors.New("no pachd address specified")
+	// DefaultPachdAddress is the default PachdAddress that should be used
+	// if none is otherwise specified. It's a loopback that should rely on
+	// port forwarding.
+	DefaultPachdAddress = PachdAddress{
+		Secured: false,
+		Host:    "0.0.0.0",
+		Port:    DefaultPachdNodePort,
+	}
+)
+
+// PachdAddress represents a parsed pachd address value
+type PachdAddress struct {
+	// Secured specifies whether grpcs should be used
+	Secured bool
+	// Host specifies the pachd address host without the port
+	Host string
+	// Port specifies the pachd port
+	Port uint16
+}
+
+// ParsePachdAddress parses a string into a pachd address, or returns an error
+// if it's invalid
+func ParsePachdAddress(value string) (PachdAddress, error) {
```

Comment:
Should this return a pointer? I just ask because that's what `getUserMachineAddrAndOpts` returns.